clarity/src/vm/ast/parser_v2/lexer/mod.rs

```diff
@@ -0,0 +1,1957 @@
+                    '\\' => '\\',
+                    '\"' => '\"',
+                    'n' => '\n',
+                    't' => '\t',
+                    'r' => '\r',
+                    '0' => '\0',
+                    _ => {
+                        self.add_diagnostic(PlacedError {
+                            span: Span {
+                                start_line: self.last_line as u32,
+                                start_column: self.last_column as u32,
+                                end_line: self.line as u32,
+                                end_column: self.column as u32,
+                            },
+                            e: LexerError::UnknownEscapeChar(self.next),
+                        });
+                        '?'
+                    }
+                };
+                s.push(ch);
+                escaped = false;
+            } else {
+                match self.next {
+                    '"' => {
+                        self.read_char();
+                        return s;
+                    }
+                    '\\' => escaped = !escaped,
+                    '\0' | '\n' => {
+                        self.add_diagnostic(PlacedError {
+                            span: Span {
+                                start_line: self.line as u32,
+                                start_column: self.column as u32,
+                                end_line: self.line as u32,
+                                end_column: self.column as u32,
+                            },
+                            e: LexerError::ExpectedClosing('"'),
+                        });
+                        self.add_diagnostic(PlacedError {
+                            span: Span {
+                                start_line,
+                                start_column,
+                                end_line: start_line,
+                                end_column: start_column,
+                            },
+                            e: LexerError::NoteToMatchThis('"'),
+                        });
+                        return s;
+                    }
+                    _ => {
+                        if !self.next.is_ascii() {
+                            self.add_diagnostic(PlacedError {
+                                span: Span {
+                                    start_line: self.line as u32,
+                                    start_column: self.column as u32,
+                                    end_line: self.line as u32,
+                                    end_column: self.column as u32,
+                                },
+                                e: LexerError::IllegalCharString(self.next),
+                            });
+                        } else {
+                            s.push(self.next);
+                        }
+                    }
+                }
+            }
+            self.read_char();
+        }
+    }
+
+    fn read_utf8_encoding(&mut self) -> String {
```

Comment:
I think this is meant to read `X` in `\u{X}`, right?  If so, can you document this?

Reply:
Added a comment in e5d99b691a270764a7b55e09ed40825cb765d3e4